assets/src/edit-story/components/canvas/canvasLayout.js

```diff
@@ -1,62 +1,223 @@
+		background-color: darkgray;
+		backface-visibility: hidden;
+		transform-style: preserve-3d;
+		transform-origin: 50% 50%;
+		transform: perspective(2500px) scale(0.4) rotateY(${ DEBUG_ANGLE }deg);
+	` ) }
+`;
+
+const Layer = css`
+	position: absolute;
+	top: 0;
+	left: 0;
+	right: 0;
+	bottom: 0;
+`;
 
+// @todo: the menu and carousel heights are not correct until we make a var-size
+// page.
+const Layout = css`
 	display: grid;
 	grid:
-    ".    .      .          . .        ." 1fr
-    ".    prev   page       . next     ." ${ PAGE_HEIGHT }px
-    ".    .      menu       . .        ." 48px
-    ".    .      carrousel  . .        ." 1fr
-    / 1fr ${ LEFT_NAV_WIDTH }px ${ PAGE_WIDTH }px ${ PAGE_NAV_PADDING }px 1fr ${ CENTRAL_RIGHT_PADDING }px;
+    ".    .      . .          . .        ." 1fr
+    ".    prev   . page       . next     ." ${ PAGE_HEIGHT }px
+    ".    .      . menu       . .        ." 48px
+    ".    .      . .          . .        ." 1fr
+    ".    carrousel      carrousel carrousel  carrousel carrousel        ." 60px
+    / 1fr ${ LEFT_NAV_WIDTH }px ${ PAGE_NAV_BUTTON_WIDTH }px ${ PAGE_WIDTH }px ${ PAGE_NAV_BUTTON_WIDTH }px ${ LEFT_NAV_WIDTH }px 1fr;
+`;
+
+const Canvas = css`
+	position: relative;
+	width: 100%;
+	height: 100%;
+`;
+
+const DisplayLayer = styled.div`
+	${ Layer }
+	${ Layout }
+
+	${ DEBUG_ANGLE && ( `
+		background-color: rgba(0, 255, 0, 0.2);
+		transform: translateZ(200px);
+	` ) }
+`;
+
+const DisplayCanvas = styled.div`
+	${ Canvas }
+	grid-area: page;
+
+	${ DEBUG_ANGLE && ( `
+		background-color: rgba(255, 255, 255, 0.2);
+	` ) }
+`;
+
+const NavLayer = styled.div`
+	${ Layer }
+	${ Layout }
+	pointer-events: none;
+
+	${ DEBUG_ANGLE && ( `
+		background-color: rgba(0, 0, 255, 0.2);
+		transform: translateZ(400px);
+	` ) }
 `;
 
-const Area = styled.div`
+const NavArea = styled.div`
+	background-color: ${ ( { theme } ) => theme.colors.bg.v1 };
 	grid-area: ${ ( { area } ) => area };
 	height: 100%;
 	width: 100%;
+	pointer-events: initial;
+
+	${ DEBUG_ANGLE && ( `
+		background-color: rgba(0, 0, 255, 0.4);
+		transform: translateZ(400px);
+	` ) }
+`;
+
+const FramesAndSelectionLayer = styled.div`
+	${ Layer }
+	${ Layout }
+	pointer-events: none;
+
+	${ DEBUG_ANGLE && ( `
+		background-color: rgba(0, 255, 255, 0.2);
+		transform: translateZ(600px);
+	` ) }
+`;
+
+const Overlay = css`
+	position: absolute;
+	top: 0;
+	left: 0;
+	width: 0;
+	height: 0;
+
+	${ DEBUG_ANGLE && ( `
+		width: 100px;
+		height: 100px;
+	` ) }
+`;
+
+const FramesOverlay = styled.div`
+	${ Overlay }
+
+	${ DEBUG_ANGLE && ( `
+		background-color: rgba(255, 0, 255, 0.2);
+	` ) }
+`;
+
+const SelectionOverlay = styled.div`
+	${ Overlay }
+
+	${ DEBUG_ANGLE && ( `
+		top: 20px;
+		left: 20px;
+		background-color: rgba(255, 255, 0, 0.2);
+	` ) }
+`;
+
+// QQQQ: only show graypane for few element types.
+const EditLayer = styled.div`
+	${ Layer }
+	${ Layout }
+	pointer-events: none;
+	background-color: rgba(0, 0, 0, 0.5);
```

Comment:
put into theme

Reply:
Do we put shading into themes? There's no actual color here, just `0.5` value that matters. This is similar to `opacity: 0.5`, but applied directly to a color. I was under an impression that those cases do not belong in a theme.

/cc @barklund @miina

Reply:
I guess "color" is a matter of definition :) I'd count black as one for our purposes?

Regardless, I think it might be worthwhile to put even `opacity` values into the theme, as they're stylistic choices (similar to fonts, font style, ...).

Reply:
Ok. Will do then.